cmd/cinder-csi-plugin/main.go

```diff
@@ -100,6 +109,20 @@ func main() {
 
 func handle() {
 
+	if httpEndpoint != "" {
+		metricsManager := metrics.NewCSIMetricsManager("" /* driverName */)
```

Comment:
@jichenjc I have same question as @lingxiankong . What kind of metrics are being supported? If its openstack api calls, we can just reuse/extending the existing endpoint , because its the same thing we want , https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/metrics/metrics_api.go

If its for CSI operations for which you have used the above library , Im confused . Is it really required? 
I see in external-provisioner and all ther sidecars they have added metrics which give metrics for all CSI operations. From this issue , it is said that https://github.com/kubernetes-csi/docs/issues/339 ,  "Driver deployments should set those flags to ensure the metrics are being emitted." , so dont we need to just set the flags in the sidecars for CSI operations?

I might have misunderstood, but could you pls explain with the examples , that could be real help. Thanks

Reply:
@ramineni @lingxiankong please check comments here for the purpose 
https://github.com/kubernetes/cloud-provider-openstack/issues/1392#issuecomment-799869304

thanks